components/support/error/src/handling.rs

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Helpers for components to "handle" errors.
+
+/// Describes what error reporting action should be taken.
+pub enum ErrorReporting {
+    /// No logging or error reporting.
+    Nothing,
+    /// We write a log message but don't report it.
+    Log { level: log::Level },
+    /// We log a message and report via our error reporter.
+    Report {
+        level: log::Level,
+        report_class: String,
+    },
+}
```

Comment:
I like how flexible this is.

You could go a step further and have a struct of options:
```
pub struct ErrorReporting {
  log: Option<log::Level>,
  report: Option<String>,
  count_in_telemetry: Option<String>, # .. eventually
}
```

I don't have a preference between the two options.  I mostly just wanted to through this out as a possibility.

Reply:
I actually like this better, thanks, especially given your `count_in_telemetry` example - the enum approach would have meant a new variant for every *combination* of actions to take.